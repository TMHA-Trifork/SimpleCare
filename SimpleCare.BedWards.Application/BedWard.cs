namespace SimpleCare.BedWards.Application;

public class BedWard;
