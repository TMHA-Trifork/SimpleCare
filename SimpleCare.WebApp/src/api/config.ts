export const API_CONFIG = {
  baseUrl: 'http://localhost:5034'
}

export function getApiUrl(path: string): string {
  return `${API_CONFIG.baseUrl}${path}`
}
