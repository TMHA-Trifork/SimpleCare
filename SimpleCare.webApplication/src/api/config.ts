export const API_CONFIG = {
  baseUrl: 'https://localhost:7023'
}

export function getApiUrl(path: string): string {
  return `${API_CONFIG.baseUrl}${path}`;
}
