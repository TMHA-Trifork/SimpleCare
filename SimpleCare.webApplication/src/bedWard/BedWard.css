.bed-ward-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  font-family: Arial, sans-serif;
}

.bed-ward-title {
  font-size: 2.5rem;
  color: #333;
  margin-bottom: 20px;
  text-shadow: 1px 1px 2px rgba(0, 0, 0, 0.2);
}

.bed-ward-table {
  width: 80%;
  max-width: 800px;
  margin: 0 auto;
  border-collapse: collapse;
  background: #fff;
  border-radius: 10px;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  overflow: hidden;
}

.bed-ward-table th,
.bed-ward-table td {
  padding: 15px;
  text-align: left;
  border-bottom: 1px solid #ddd;
  font-size: 1.2rem;
  color: #555;
}

.bed-ward-table th {
  background: #d4d4d4;
  font-weight: bold;
  color: #333;
}

.bed-ward-table tr:last-child td {
  border-bottom: none;
}

.bed-ward-row:hover {
  background: #f0f8ff;
}
