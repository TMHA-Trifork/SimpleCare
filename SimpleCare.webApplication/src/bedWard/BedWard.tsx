import { useEffect, useState } from "react";
import {
  fetchBedWardPatients,
  BedWardPatientListItem,
} from "../api/bedWardPatients";
import "./BedWard.css";

export const BedWard = () => {
  const [patients, setPatients] = useState<BedWardPatientListItem[]>([]);

  useEffect(() => {
    const loadPatients = async () => {
      try {
        const data = await fetchBedWardPatients();
        setPatients(data);
      } catch (error) {
        console.error("Error fetching bed ward patients:", error);
      }
    };

    loadPatients();
  }, []);

  return (
    <div className="bed-ward-container">
      <h1 className="bed-ward-title">Bed Ward</h1>
      <table className="bed-ward-table">
        <thead>
          <tr>
            <th>Given Names</th>
            <th>Family Name</th>
            <th>Status</th>
          </tr>
        </thead>
        <tbody>
          {patients.map((patient, index) => (
            <tr key={index} className="bed-ward-row">
              <td>{patient.givenNames}</td>
              <td>{patient.familyName}</td>
              <td>{patient.personalIdentifier}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};
