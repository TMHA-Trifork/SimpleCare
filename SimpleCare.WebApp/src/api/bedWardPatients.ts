import { components } from '../types/api-types'
import { getApiUrl } from './config'

export type BedWardPatientListItem = components['schemas']['BedWardPatientListItem']
export type BedWardPatient = components['schemas']['BedWardPatient']
export type PatientAdmission = components['schemas']['PatientAdmission']

export async function fetchBedWardPatients(): Promise<BedWardPatientListItem[]> {
  const response = await fetch(getApiUrl('/api/bed-ward/patients'))
  if (!response.ok) {
    throw new Error('Network response was not ok')
  }
  return response.json()
}

export async function fetchBedWardPatient(patientId: string): Promise<BedWardPatient> {
  const response = await fetch(getApiUrl(`/api/bed-ward/patients/${patientId}`))
  if (!response.ok) {
    throw new Error('Network response was not ok')
  }
  return response.json()
}

export async function admitPatient(admission: PatientAdmission): Promise<void> {
  const response = await fetch(getApiUrl('/api/bed-ward/patients/admit'), {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(admission)
  })
  if (!response.ok) {
    throw new Error('Network response was not ok')
  }
}
