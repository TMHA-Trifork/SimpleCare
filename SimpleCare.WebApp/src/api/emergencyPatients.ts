import { components } from "../types/api-types";
import { getApiUrl } from "./config";

export type EmergencyPatientListItem =
  components["schemas"]["EmergencyPatientListItem"];
export type EmergencyPatient = components["schemas"]["EmergencyPatient"];
export type EmergencyRegistration =
  components["schemas"]["EmergencyRegistration"];
export type TransferRequest = components["schemas"]["TransferRequest"];

export async function fetchEmergencyPatients(
  status?: components["schemas"]["EmergencyPatientStatus"][]
): Promise<EmergencyPatientListItem[]> {
  const url = new URL(getApiUrl("/api/emergency-wards/patients"));
  if (status && status.length > 0) {
    status.forEach((s) => url.searchParams.append("status", s));
  }
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error("Network response was not ok");
  }
  return response.json();
}

export async function fetchEmergencyPatient(
  patientId: string
): Promise<EmergencyPatient> {
  const response = await fetch(
    getApiUrl(`/api/emergency-wards/patients/${patientId}`)
  );
  if (!response.ok) {
    throw new Error("Network response was not ok");
  }
  return response.json();
}

export async function registerEmergencyPatient(
  registration: EmergencyRegistration
): Promise<void> {
  const response = await fetch(
    getApiUrl("/api/emergency-wards/patients/register"),
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(registration),
    }
  );
  if (!response.ok) {
    throw new Error("Network response was not ok");
  }
}

export async function transferEmergencyPatient(
  transfer: TransferRequest
): Promise<void> {
  const response = await fetch(
    getApiUrl("/api/emergency-wards/patients/transfer"),
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(transfer),
    }
  );
  if (!response.ok) {
    throw new Error("Network response was not ok");
  }
}
