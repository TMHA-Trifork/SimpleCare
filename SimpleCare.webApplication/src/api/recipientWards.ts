import { getApiUrl } from "./config";

export type RecipientWard = {
  id: string;
  name: string;
};

export async function fetchRecipientWards(): Promise<RecipientWard[]> {
  const response = await fetch(
    getApiUrl("/api/emergency-wards/recipient-wards")
  );
  if (!response.ok) {
    throw new Error("Network response was not ok");
  }
  return response.json();
}
