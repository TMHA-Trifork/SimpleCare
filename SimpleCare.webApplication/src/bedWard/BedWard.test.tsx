import { describe, it, expect, vi, Mock } from "vitest";
import { renderWithoutRouter, act } from "../test/test-utils.tsx";
import { BedWard } from "./BedWard";
import * as api from "../api/bedWardPatients";

vi.mock("../api/bedWardPatients");

const mockFetchBedWardPatients = api.fetchBedWardPatients as Mock;

describe("BedWard", () => {
  it("should match snapshot", async () => {
    const mockPatients = [
      {
        patientId: "1",
        givenNames: "John",
        familyName: "Doe",
        personalIdentifier: "123456",
        roomNumber: "101",
      },
    ];

    mockFetchBedWardPatients.mockResolvedValueOnce(mockPatients);

    let container: HTMLElement;
    await act(async () => {
      const result = renderWithoutRouter(<BedWard />);
      container = result.container;
    });

    expect(container!.innerHTML).toMatchSnapshot();
  });

  it("should display patient list", async () => {
    const mockPatients = [
      {
        patientId: "1",
        givenNames: "John",
        familyName: "Doe",
        personalIdentifier: "123456",
        roomNumber: "101",
      },
    ];

    mockFetchBedWardPatients.mockResolvedValueOnce(mockPatients);

    let container: HTMLElement;
    await act(async () => {
      const result = renderWithoutRouter(<BedWard />);
      container = result.container;
    });

    expect(container!.textContent).toContain("John");
    expect(container!.textContent).toContain("Doe");
    expect(container!.textContent).toContain("123456");
  });
});
