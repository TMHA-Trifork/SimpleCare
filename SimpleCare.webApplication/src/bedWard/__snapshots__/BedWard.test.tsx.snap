// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BedWard > should match snapshot 1`] = `"<div class="bed-ward-container"><h1 class="bed-ward-title">Bed Ward</h1><table class="bed-ward-table"><thead><tr><th>Given Names</th><th>Family Name</th><th>Status</th></tr></thead><tbody><tr class="bed-ward-row"><td>John</td><td>Doe</td><td>123456</td></tr></tbody></table></div>"`;
